### PR TITLE
Issue 219 - Create shopping cart API

### PR DESCRIPTION
**Pull request by javier-gomez-github**
- Added new BaseEntitlementWithAddOnsSpecifier to group multiple bundles.
- Added method to the EntitlementApi to create multiple entitlements with Add Ons.

### CODE REVIEW

File: src/main/java/org/killbill/billing/entitlement/api/BaseEntitlementWithAddOnsSpecifier.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import java.util.UUID;
+
+import org.joda.time.LocalDate;
+
+public interface BaseEntitlementWithAddOnsSpecifier {
+
+    /**
+     * @return the bundle Id
+     */
+    public UUID getBundleId();
```
Comment (pierre) [inline src/main/java/org/killbill/billing/entitlement/api/BaseEntitlementWithAddOnsSpecifier.java]: Do you need this?

Comment (javier-gomez-github) [inline src/main/java/org/killbill/billing/entitlement/api/BaseEntitlementWithAddOnsSpecifier.java]: The `bundleId`, `externalKey`, `entitlementEffectiveDate` and `billingEffectiveDate` were part of the `DefaultEntitlementContext`. 
Now all those attributes are part of the `BaseEntitlementWithAddOnsSpecifier`.
Do you think it should not be there?

Comment (sbrossie) [inline src/main/java/org/killbill/billing/entitlement/api/BaseEntitlementWithAddOnsSpecifier.java]: @javier-gomez-github I am not sure exactly why @pierre made that comment. He is at at a conference tight now, so unclear if he will have time to comment. I would keep it the way it is for now, and we can always revisit.

Comment (pierre) [inline src/main/java/org/killbill/billing/entitlement/api/BaseEntitlementWithAddOnsSpecifier.java]: No problem in having it. When we designed the API on the main ticket, we didn't have it - I just wanted to make sure it was indeed an oversight on our end.

**Pull request merged**